packages/bddeb

```diff
@@ -140,13 +171,52 @@ def get_parser():
 
     parser.add_argument("--signuser", default=False, action='store',
                         help="user to sign, see man dpkg-genchanges")
+
+    parser.add_argument("--packaging-branch", nargs="?", metavar="BRANCH",
+                        const="ubuntu/devel", type=str,
+                        help=(
+                            "Import packaging from %(metavar)s instead of"
+                            " using the packages/debian/* templates"
+                            " (default: %(const)s)"
+                        ))
+
     return parser
 
 
 def main():
     parser = get_parser()
     args = parser.parse_args()
 
+    if args.packaging_branch:
+        try:
+            subp.subp(
+                [
+                    "git",
+                    "show-ref",
```

Comment:
This will only match/succeed if you actually have a local copy of the local branch checked out. Shouldn't we be checking upstream/args.packaging_branch ?

Reply:
(You mean "a local copy of the *packaging* branch checked out" I guess.)

You are right, and the `git archive <branch> debian` command will also fail if `<branch>` (usually `ubuntu/devel`) is not checked out locally. We could try to do it automatically, but there are two caveats:
 * We don't know what's the remote the user wants to use. We could add heuristics at cost of extra complexity, probably not worth it.
 * I wouldn't like a "build package" tool to touch my git repo.

I'm thinking of just hinting the user to checkout the packaging branch in the error message. It's an easy step to do and also makes very clear where the packaging bits are coming from. What do you think?

Reply:
I think the hinting sounds good.